src/systems/advanced_lift_drag/AdvancedLiftDrag.cc

```diff
@@ -0,0 +1,840 @@
+      this->validConfig = false;
+      return;
+    }
+    else if (entities.size() > 1)
+    {
+      gzwarn << "Multiple link entities with name[" << linkName << "] found. "
+             << "Using the first one.\n";
+    }
+
+    this->linkEntity = *entities.begin();
+    if (!_ecm.EntityHasComponentType(this->linkEntity,
+                                     components::Link::typeId))
+    {
+      this->linkEntity = kNullEntity;
+      gzerr << "Entity with name[" << linkName << "] is not a link.\n";
+      this->validConfig = false;
+      return;
+    }
+  }
+  else
+  {
+    gzerr << "AdvancedLiftDrag system requires the 'link_name' parameter.\n";
+    this->validConfig = false;
+    return;
+  }
+
+  // If we reached here, we have a valid configuration
+  this->validConfig = true;
+}
+
+//////////////////////////////////////////////////
+AdvancedLiftDrag::AdvancedLiftDrag()
+    : System(), dataPtr(std::make_unique<AdvancedLiftDragPrivate>())
+{
+}
+
+//////////////////////////////////////////////////
+void AdvancedLiftDragPrivate::Update(EntityComponentManager &_ecm)
+{
+  GZ_PROFILE("AdvancedLiftDragPrivate::Update");
+  // get linear velocity at cp in world frame
+  const auto worldLinVel =
+      _ecm.Component<components::WorldLinearVelocity>(this->linkEntity);
+  const auto worldAngVel =
+      _ecm.Component<components::WorldAngularVelocity>(this->linkEntity);
+  const auto worldPose =
+      _ecm.Component<components::WorldPose>(this->linkEntity);
+
+
+  std::vector<components::JointPosition*> controlJointPosition_vec(
+    this->num_ctrl_surfaces);
+  // Generate a new vector that contains the current positions for all joints
+  for(int i = 0; i < this->num_ctrl_surfaces; i++){
+    components::JointPosition *controlJointPosition = nullptr;
+    if(this->controlJoints[i] != kNullEntity){
+      controlJointPosition = _ecm.Component<components::JointPosition>(
+        this->controlJoints[i]);
+      controlJointPosition_vec[i] = controlJointPosition;
+    }
+  }
+
+  if (!worldLinVel || !worldAngVel || !worldPose)
+    return;
+
+  const auto &pose = worldPose->Data();
+  const auto cpWorld = pose.Rot().RotateVector(this->cp);
+  const auto air_velocity = worldLinVel->Data() + worldAngVel->Data().Cross(
+    cpWorld);
+
+  // Define body frame: X forward, Z downward, Y out the right wing
+  gz::math::Vector3d body_x_axis = pose.Rot().RotateVector(this->forward);
+  gz::math::Vector3d body_z_axis = -1*(pose.Rot().RotateVector(this->upward));
+  gz::math::Vector3d body_y_axis =  body_z_axis.Cross(body_x_axis);
+
+  // Get the in-plane velocity (remove spanwise velocity from the velocity
+  // vector air_velocity)
+  gz::math::Vector3d velInLDPlane = air_velocity - air_velocity.Dot(
+    body_y_axis)*body_y_axis;
+
+  // Compute dynamic pressure
+  if(std::fabs(velInLDPlane.Length()) <= 1e-9){
+    gzerr << "In-plane velocity of vehicle cannot be 0.\n";
+    this->validConfig = false;
```

Comment:
```suggestion
    
```

The plane can be at rest. For instance when I start the example, the plane usually has zero velocity. It seems returning is the right behaviour here. There is no need to mark the config as invalid or log an error.

Reply:
but we still need a check for `speedinLDPlane` not being zero.

Reply:
That is correct.

Reply:
@arjo129 I have amended this in the latest commit.